File: backend/api/routes/banner.js
const express = require('express');
const router = express.Router();
const { 
  getBanners, 
  getBannerById, 
  createBanner, 
  updateBanner, 
  deleteBanner 
} = require('../controllers/bannerController');
const { uploadBanner } = require('../middleware/uploadMiddleware');
const { authMiddleware, roleMiddleware } = require('../middleware');

// Public routes
router.get('/', getBanners);
router.get('/:id', getBannerById);

// Admin routes
router.post('/', authMiddleware, roleMiddleware('admin'), uploadBanner, createBanner);
router.put('/:id', authMiddleware, roleMiddleware('admin'), uploadBanner, updateBanner);
router.patch('/:id', authMiddleware, roleMiddleware('admin'), uploadBanner, updateBanner);
router.delete('/:id', authMiddleware, roleMiddleware('admin'), deleteBanner);

module.exports = router;

File: fastapi/src/routes/banner.py
from fastapi import APIRouter
from src.controllers.banner_controller import get_banners, get_banner_by_id

router = APIRouter()


@router.get("/", response_model=dict)
async def get_banners_route():
    return await get_banners()


@router.get("/{banner_id}", response_model=dict)
async def get_banner_route(banner_id: str):
    return await get_banner_by_id(banner_id)